std/typecons.d

```diff
@@ -51,12 +51,12 @@ Encapsulates unique ownership of a resource.  Resource of type $(D T) is
 deleted at the end of the scope, unless it is transferred.  The
 transfer can be explicit, by calling $(D release), or implicit, when
 returning Unique from a function. The resource can be a polymorphic
-class object, in which case Unique behaves polymorphically too.
+class object or instance of an interface, in which case Unique behaves polymorphically too.
 */
 struct Unique(T)
 {
 /** Represents a reference to $(D T). Resolves to $(D T*) if $(D T) is a value type. */
-static if (is(T:Object))
+static if (is(T == class) || is(T == interface))
```

Comment:
I don't know a lot about `extern(c++)` or `extern(Windows)`, but from readinghttp://dlang.org/spec/interfaceToC.html, it seems like `extern(Windows)` isn't really relevant since it is more about function calling conventions, and from http://dlang.org/spec/cpp_interface.html it seems like this might work with C++ classes. Although it isn't really clear if it is safe to call `destroy` on a C++ class. Of course, Unique certainly won't clean up the memory for a C++ class, but that is true regardless of if `RefT` is `T` or `T*`